File: backend/app/schemas/rule.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional

class RuleBase(BaseModel):
    rule_text: str
    is_active: bool = True

class RuleCreate(RuleBase):
    repository_id: int

class RuleUpdate(BaseModel):
    rule_text: Optional[str] = None
    is_active: Optional[bool] = None

class CustomRule(RuleBase):
    id: int
    repository_id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/rule.ts
export interface CustomRule {
  id: number;
  repository_id: number;
  rule_text: string;
  is_active: boolean;
  created_at: string;
}

export interface CreateCustomRuleInput {
  repository_id: number;
  rule_text: string;
  is_active: boolean;
}

export interface UpdateCustomRuleInput {
  rule_text?: string;
  is_active?: boolean;
}
